pyop2/base.py

```diff
@@ -2218,30 +2030,52 @@ def __itruediv__(self, other):
     __idiv__ = __itruediv__  # Python 2 compatibility
 
     @collective
-    def halo_exchange_begin(self, reverse=False):
-        """Begin halo exchange.
+    def global_to_local_begin(self, insert_mode):
+        """Begin a halo exchange from global to ghosted representation.
+
+        :kwarg insert_mode:  insertion mode (an access descriptor)"""
+        halo = self.dataset.halo
+        if halo is None:
+            return
+        if insert_mode in [READ, RW] and not self.halo_valid:
+            halo.global_to_local_begin(self, WRITE)
```

Comment:
I realise the `insert_mode` name changes meaning from this method to the halo call.  To the halo it means, do we write or sum into the array.  Here, insert_mode is actually access_mode.